src/wp-admin/includes/image-edit.php

```diff
@@ -799,6 +801,35 @@ function wp_restore_image( $post_id ) {
 		}
 	}
 
+	// Get the next available `full-{orig or hash}` key on the images if the name
+	// has not been used as part of the backup sources it would be used if no size is
+	// found or backup exists `null` would be used instead.
+	$next_full_size_key_from_backup = null;
+	foreach ( array_keys( $backup_sizes ) as $size_name ) {
+		// If the target already has the sources attributes find the next one.
+		if ( isset( $backup_sources[ $size_name ] ) ) {
+			continue;
+		}
+
+		// We are only interested in the `full-` sizes.
+		if ( strpos( $size_name, 'full-' ) === false ) {
+			continue;
+		}
+
+		$next_full_size_key_from_backup = $size_name;
```

Comment:
We break the loop when we get the first instance so I think to update the meta here.

```suggestion
                $backup_sources[ $size_name ] = $old_meta['sources'];
		update_post_meta( $post_id, '_wp_attachment_backup_sources', $backup_sources );
```

Reply:
Added to the code!